File: VB/ImplementCustomUserBasedOnExistingClassEF.Module/BusinessObjects/ImplementCustomUserBasedOnExistingClassEFDbContext.vb
Imports System
Imports System.Data
Imports System.Linq
Imports System.Data.Entity
Imports System.Data.Common
Imports System.Data.Entity.Core.Objects
Imports System.Data.Entity.Infrastructure
Imports System.ComponentModel
Imports DevExpress.ExpressApp.EF.Updating
Imports DevExpress.Persistent.BaseImpl.EF
Imports DevExpress.Persistent.BaseImpl.EF.PermissionPolicy

Namespace ImplementCustomUserBasedOnExistingClassEF.Module.BusinessObjects
    Public Class ImplementCustomUserBasedOnExistingClassEFDbContext
        Inherits DbContext

        Public Sub New(ByVal connectionString As String)
            MyBase.New(connectionString)
        End Sub
        Public Sub New(ByVal connection As DbConnection)
            MyBase.New(connection, False)
        End Sub
        Public Property ModulesInfo() As DbSet(Of ModuleInfo)
        Public Property Roles() As DbSet(Of PermissionPolicyRole)
        Public Property TypePermissionObjects() As DbSet(Of PermissionPolicyTypePermissionObject)
        Public Property Users() As DbSet(Of PermissionPolicyUser)
        Public Property ModelDifferences() As DbSet(Of ModelDifference)
        Public Property ModelDifferenceAspects() As DbSet(Of ModelDifferenceAspect)
        Public Property Employees() As DbSet(Of Employee)
        Public Property EmployeeRoles() As DbSet(Of EmployeeRole)
        Public Property EmployeeTasks() As DbSet(Of EmployeeTask)
    End Class
End Namespace

File: CS/ImplementCustomUserBasedOnExistingClassEF.Module/BusinessObjects/ImplementCustomUserBasedOnExistingClassEFDbContext.cs
using System;
using System.Data;
using System.Linq;
using System.Data.Entity;
using System.Data.Common;
using System.Data.Entity.Core.Objects;
using System.Data.Entity.Infrastructure;
using System.ComponentModel;
using DevExpress.ExpressApp.EF.Updating;
using DevExpress.Persistent.BaseImpl.EF;
using DevExpress.Persistent.BaseImpl.EF.PermissionPolicy;

namespace ImplementCustomUserBasedOnExistingClassEF.Module.BusinessObjects {
	public class ImplementCustomUserBasedOnExistingClassEFDbContext : DbContext {
		public ImplementCustomUserBasedOnExistingClassEFDbContext(String connectionString)
			: base(connectionString) {
		}
		public ImplementCustomUserBasedOnExistingClassEFDbContext(DbConnection connection)
			: base(connection, false) {
		}
		public DbSet<ModuleInfo> ModulesInfo { get; set; }
	    public DbSet<PermissionPolicyRole> Roles { get; set; }
		public DbSet<PermissionPolicyTypePermissionObject> TypePermissionObjects { get; set; }
		public DbSet<PermissionPolicyUser> Users { get; set; }
		public DbSet<ModelDifference> ModelDifferences { get; set; }
        public DbSet<ModelDifferenceAspect> ModelDifferenceAspects { get; set; }
        public DbSet<Employee> Employees { get; set; }
        public DbSet<EmployeeRole> EmployeeRoles { get; set; }
        public DbSet<EmployeeTask> EmployeeTasks { get; set; }
	}
}